src/Security/PasswordValidator.php

```diff
@@ -3,52 +3,65 @@
 namespace SilverStripe\Security;
 
 use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Extensible;
 use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Dev\Deprecation;
 use SilverStripe\ORM\ValidationResult;
 
 /**
  * This class represents a validator for member passwords.
- *
- * <code>
- * $pwdVal = new PasswordValidator();
- * $pwdValidator->minLength(7);
- * $pwdValidator->checkHistoricalPasswords(6);
- * $pwdValidator->characterStrength(3, array("lowercase", "uppercase", "digits", "punctuation"));
- *
- * Member::set_password_validator($pwdValidator);
- * </code>
  */
 class PasswordValidator
 {
     use Injectable;
     use Configurable;
+    use Extensible;
 
     /**
      * @config
      * @var array
      */
-    private static $character_strength_tests = array(
+    private static $character_strength_tests = [
         'lowercase' => '/[a-z]/',
         'uppercase' => '/[A-Z]/',
         'digits' => '/[0-9]/',
         'punctuation' => '/[^A-Za-z0-9]/',
-    );
+    ];
 
-    protected $minLength, $minScore, $testNames, $historicalPasswordCount;
+    /**
+     * @config
+     * @var integer
+     */
+    private static $min_length = 3;
+
+    /**
+     * @config
+     * @var integer
+     */
+    private static $min_test_score = null;
+
+    /**
+     * @config
+     * @var integer
+     */
+    private static $historic_count = 6;
 
     /**
+     * @deprecated 5.0
      * Minimum password length
      *
      * @param int $minLength
      * @return $this
      */
     public function minLength($minLength)
     {
-        $this->minLength = $minLength;
+        Deprecation::notice('5.0', 'Use ->config()->set(\'min_length\', $value) instead.');
```

Comment:
You've deprecated these in 4.x via https://github.com/silverstripe/silverstripe-framework/pull/7800 - should these be removals in the master branch instead?

Reply:
I'd remove them in 5.1 as a preference - 5.0 means that the method is highlighted with a notice
But yeah could swing either way.